NTUEELightDance/LightDance-Editor
# EDITOR-#459

### What is this PR for?
Read a music from files/music and generate a waveform png file to the same location

### What type of PR is it?

[Feature]

### Todos


### What is the Github issue?
resolve #459

### How should this be tested?

Go to utils, and install all the dependencies listed in package.json by **pnpm install**.
Note that the "sharp" module has to be version 0.32.6. The latest version might cause some problem.
Then run 

```
pnpm generateWaveform
```
This should read 2023.mp3 and output waveform.png in the same folder.
The route of the files is currently hard coded in the beginning of the program, and it can be altered to test other files.
I tested in linux system and not sure if it would work for mac qq.


### Screenshots (if appropriate)

### Questions

* Do the license files need updating? Yes/No
* Are there breaking changes for older versions? Yes/No
* Does this need new documentation? Yes/No
